### ISSUE TITLE
Support making Android wear apps

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#2057]

**Comment by soundanalogous** [comment]
Is there any reason this Flutter apps would not build for Wear? Seems they should since you can build for android. I'll give it a try soon (since I'm a UXE on the Wear team).

**Comment by eseidelGoogle** [comment]
I am pretty sure @tonygentilcore ran flutter on a watch many moons ago.  It's just not a workflow we've tried to support yet.

**Comment by soundanalogous** [comment]
You can determine if the screen is round by the Android [WindowInsets.isRound()](http://developer.android.com/reference/android/view/WindowInsets.html#isRound()) method. It may make sense to expose that in the [WindowPadding class](http://docs.flutter.io/flutter/dart-ui/WindowPadding-class.html).

**Comment by abarth** [comment]
Yeah, we should add a `shape` property to `Window` that's something like `BoxShape`.

**Comment by jaumard** [comment]
Andrea wear (and watch OS) would be awesome, but generally phone app and wear app are dialoging to exchange or stock data. How it will work if both phone and wear app are in flutter ? In native we use the Api to exchange data or messages but I'm curious how it will be handle by flutter as the phone app may not be launched when the wear app is asking for data

**Comment by jaumard** [comment]
@Hixie someone made a package already to offer some widgets dedicated to android wear https://pub.dartlang.org/packages/wear

**Comment by AndroidNils** [comment]
There is a way to develop wearOS apps: https://medium.com/flutter-community/flutter-building-wearos-app-fedf0f06d1b4

**Comment by jaumard** [comment]
@AndroidNils I knew that way but by doing like this you can't make a companion app as you sacrifice the android app to be an android wear app, you can't have both

**Comment by arnaudelub** [comment]
well, it could be, just need to add Platform.isWearOs right?

**Comment by au-top** [comment]
wearOs  InputDecoration  The pop-up keyboard is at the top, not the bottom 

```

https://github.com/flutter/flutter/issues/70068
https://github.com/flutter/flutter/issues/24001

```

**Comment by vladislav1nikov** [comment]
Any news on WearOS support?

**Comment by vladyslav-nikov** [comment]
What are your plans for WearOS?

**Comment by Gabriel2048** [comment]
Hello team.
I see this issue has a very low priority (P6)

However with the rapid adoptions of wearables and Google's recent efforts towards the wearOS platform this will give an incentive for developers to consider and build wearOS apps and having Flutter first party support will make Flutter more appealing to a broader range of developers and organizations.
Given the fact that numerous third party Flutter plugins attempts([wear](https://pub.dev/packages/wear) [flutter_wear](https://pub.dev/packages/flutter_wear) [dotup_flutter_wear](https://pub.dev/packages/dotup_flutter_wear) [is_wear](https://pub.dev/packages/is_wear/example) [watch_connectivity](https://pub.dev/packages/watch_connectivity)) were tried and numerous stackoverflow questions and Medium articles exists this is a clear indication there is a market for this and developers are working with sub-optimal tools.

Having this support will also align better with Flutter's slogan "Build apps for any screen".

**Comment by Mstrodl** [comment]
It's worth noting that most of these addons are pretty outdated. I've been working on trying to update the `wear` package to use the latest WearOS APIs and am running into some issues that would be much more easily solved on flutter's end (Modern ambient mode support requires ideally making changes to `FlutterFragmentActivity`. Users can work around it by modifying their `MainActivity`, but it certainly won't work as well as a native solution)

**Comment by jan-tosovsky-cz** [comment]
The Wear OS user interface supports several "surfaces" https://developer.android.com/training/wearables/user-interfaces so it would be handy to have their counterpart in Flutter. It is not only more convenient for DEVs but also beneficial for users as they get a unified UI with more predictable app behavior.

**Comment by ihavezerohealth** [comment]
I've been making a messaging application on Wear OS. There is very little support, but it does seem to be possible.
Personally, I believe that the ability to make Wear OS apps using Flutter is there already, it's just very difficult for developers as there is little support and no real documentation for using Flutter with Wear OS.

**Comment by renancaraujo** [comment]
What is the level of support the team have in mind to close this issue?

Given that:
- with a few changes in some files under `android`, theoretically, it is possible to make futter app can that runs on WearOS.
- the fact that the community has already created some articles, examples, and plugins for WearOS

 From the framework standpoint, I think Flutter support for WearOS already counts as something "production" worthy.

**Comment by rekire** [comment]
Out of my view you can write flutter apps for WearOS, but it is just very painful.

The existing WearOS plugins are not maintained (which is a overall problem with flutter plugins out of my view).

Might be a good start point would be to add some basic detection of WearOS in the safe area widget, however how to declare which part of the screen is not rendered is a bit hard to mange since it could be round.

I would like to see some curved text widget. There is [`elliptic_text`](https://pub.dev/packages/elliptic_text) but it is hard to use ether since it has some wired behaviors. I had to hack a lot with clipping and sized boxes to make it working like I would expect.

Radius scroll bars would be nice too other wise the user cannot guess the current position of the list.

For my App I have in mind to have some edge swipe sheets to (contextual) overlay my content, but there is absolutely no support in Flutter. I am aware of the `DrawerController`, `GestureDetector` and the `AnimationController` of the `BottomSheet`, but those are all just parts which cannot be reused to build what I aim. I need to copy&paste parts of those classes to a new widget to combine the features. I'm referring things like the velocity of the fling while opening the sheets.

Out of my view the biggest issue is that you need to shink flutter. When you put the support for all 4 architectures you WearOS app will be at least 70MB huge. I am not really sure why and when, but the compression of the so files in the apks have been disabled which makes the APK bigger ether. (I can guess that this might be a optimization for memory mapping the so files). That makes the downloads slower and I am not sure if they are compressed while the transmission.

**Comment by masus04** [comment]
The main features that are missing from my perspective are as follows:
* [Rotary input](https://developer.android.com/training/wearables/user-input/rotary-input) for scrolling (Required by the playstore for any scrollable widget)
* [Ongoing Activity](https://developer.android.com/training/wearables/notifications/ongoing-activity) (Required by playstore for any app that has an ongoing activity such as a timer, which are most of them)
* Location (The widely used packages for location services for some reason do not work on wearOS and the package developers do not want to support it as long as wearOS is not officially supported by Flutter)

The first two missing features (rotary input & ongoing activity) prevent many potentially working apps from being deployed on the play store while the third (location) prevents many more from even working.
